{{cookiecutter.project_name}}/.mlops-setup-scripts/cicd/main-azure.tf

```diff
@@ -38,34 +38,77 @@ provider "databricks" {
   token = module.azure_create_sp.prod_service_principal_aad_token
 }
 
-
 module "staging_workspace_cicd" {
   source = "./common"
   providers = {
     databricks = databricks.staging_sp
   }
+  {%- if cookiecutter.cicd_platform == "gitHub" %}
   git_provider    = var.git_provider
   git_token       = var.git_token
-  github_repo_url = var.github_repo_url
   env             = "staging"
+  github_repo_url = var.github_repo_url
+  {%- elif cookiecutter.cicd_platform == "azureDevOpsServices" %}
+  git_provider = var.git_provider
+  git_token    = var.git_token
+  {%- endif %}
 }
 
 module "prod_workspace_cicd" {
   source = "./common"
   providers = {
     databricks = databricks.prod_sp
   }
+  {%- if cookiecutter.cicd_platform == "gitHub" %}
   git_provider    = var.git_provider
   git_token       = var.git_token
-  github_repo_url = var.github_repo_url
   env             = "prod"
+  github_repo_url = var.github_repo_url
+  {%- elif cookiecutter.cicd_platform == "azureDevOpsServices" %}
+  git_provider = var.git_provider
+  git_token    = var.git_token
+  {%- endif %}
+}
+
+{% if cookiecutter.cicd_platform == "azureDevOpsServices" -%}
+// Additional steps for Azure DevOps. Create staging and prod service principals for an enterprise application.
+data "azuread_client_config" "current" {}
+
+resource "azuread_application" "{{cookiecutter.project_name}}-aad" {
+  display_name = "{{cookiecutter.project_name}}"
+  owners       = [data.azuread_client_config.current.object_id]
+}
+
+resource "azuread_service_principal" "staging_service_principal" {
+  application_id               = module.azure_create_sp.staging_service_principal_application_id
+  app_role_assignment_required = false
+  owners                       = [data.azuread_client_config.current.object_id]
+
+  feature_tags {
+    enterprise = true
```

Comment:
Out of curiosity, do these actually need to be enterprise applications / do we want to be opinionated about that? Or could we keep things simple/have them just be linked Azure AD service principals to start with?

Reply:
service principal == enterprise application. Only with them you can assign permissions, etc. Otherwise it's just an application, not service principal